default_warehouse_from_sale_team/models/default_warehouse_mixin.py

```diff
@@ -0,0 +1,70 @@
+from odoo import api, models
+
+
+class DefaultWarehouseMixin(models.AbstractModel):
+    """If you inherit from this model and add a field called warehouse_id into
+    the model, then the default value for such model will be the one
+    set into the sales team.
+
+    Make sure to put this mixin at the top of inheritance (before the base model), e.g.
+
+        _inherit = ["default.warehouse.mixin", "sale.order"]
+    """
+
+    _name = "default.warehouse.mixin"
+    _description = "Default Warehouse"
+
+    @api.model
+    def default_get(self, fields_list):
+        """Force that if model has a field called warehouse_id the default
+        value is the one set in the user's sales team.
+        """
+        defaults = super().default_get(fields_list)
+        default_warehouse = self.env.user._get_default_warehouse_id()
+        if not default_warehouse:
+            return defaults
+        warehouse_fields = (
+            self.env["ir.model.fields"]
+            .sudo()
+            .search(
+                [
+                    ("model", "=", self._name),
+                    ("relation", "=", "stock.warehouse"),
+                    ("ttype", "=", "many2one"),
+                    ("name", "in", fields_list),
+                ]
+            )
+        )
+        defaults.update({fld.name: default_warehouse.id for fld in warehouse_fields})
+        return defaults
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        """Pass sales team by context so it's taken into account when computing sequence name"""
+        for vals in vals_list:
+            salesteam = self._get_salesteam_from_vals(vals)
+            if salesteam:
+                self = self.with_context(sequence_salesteam_id=salesteam.id)
+                break
```

Comment:
Problem with this is, if different records would have different sales teams, the 1st found one will be used for all created records

Reply:
Yes, but it is similar as https://github.com/vauxoo-dev/addons-vauxoo/blob/dc61242f20182f44fda70128fd5bae1d82d16cf0/default_warehouse_from_sale_team/models/purchase_requisition.py#L14

Reply:
No, because that method [always gets a single record](https://github.com/odoo/odoo/blob/e8c2b0ab83135dc3392854982c3d23e217cc416d/addons/purchase_requisition/models/purchase_requisition.py#L117)

Reply:
Yes, but it could more sale_team_ids in the warehouse_id, that's the reason they use the [:1]

Reply:
I applied a first approach to what we talked about this comment, any recommendation is welcome!